src/main.cc

```diff
@@ -561,6 +561,9 @@ tarantool_free(void)
 	box_lua_free();
 	/* Lua may have reference to engine tuples. */
 	tarantool_lua_free();
+	/* Free and disable memtx tx manager first - box shouldn't use it. */
+	memtx_tx_manager_free();
+	memtx_tx_manager_use_mvcc_engine = false;
```

Comment:
Don't think this make much sense. AFAIU the issue is not with mvcc subsystem itself but with the indexes are in invalid state.

Reply:
> Don't think this make much sense. AFAIU the issue is not with mvcc subsystem itself but with the indexes are in invalid state.

Do you mean setting `memtx_tx_manager_use_mvcc_engine` to `false`? If yes, I can omit it - it was done just to make sure that MVCC is not used after shutdown.

Reply:
Completely rewrote the commit - please take a look again.
The new commit name is `memtx: free memtx mvcc before space cache`